### PR TITLE
Replace life cycle with lifecycle for consistency

### PR DESCRIPTION
**Pull request by bangelic**
Throughout several areas of some guides, "lifecycle" is spelled out as
"life cycle" and also as "lifecycle". Since the
Satellite UI uses
"lifecycle" and that is the accepted spelling for most of the software
community, all instances of "life cycle" have been updated to
lifecycle. This also required a change to module names as well as
section titles and anchors.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* [x] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [x Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8; orcharhino 6.3 on EL7/8)
* We do not accept PRs for Foreman older than 3.1.

### CODE REVIEW

File: guides/common/modules/con_introduction-to-application-lifecycle.adoc [hunk 1]
```diff
@@ -1,34 +1,34 @@
 [id="introduction_to_application_life_cycle_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/con_introduction-to-application-lifecycle.adoc:1]:
```suggestion
[id="introduction_to_application_lifecycle_{context}"]
```

File: guides/common/modules/proc_changing-the-lifecycle-environment-or-content-view-of-a-host-collection.adoc [hunk 2]
```diff
@@ -1,13 +1,13 @@
 [id="Changing_the_Life_Cycle_Environment_or_Content_View_of_a_Host_Collection_{context}"]
 = Changing the Life Cycle Environment or Content View of a Host Collection
```
Comment (maximiliankolb) [inline guides/common/modules/proc_changing-the-lifecycle-environment-or-content-view-of-a-host-collection.adoc:2]:
```suggestion
[id="Changing_the_Lifecycle_Environment_or_Content_View_of_a_Host_Collection_{context}"]
= Changing the Lifecycle Environment or Content View of a Host Collection
```

File: guides/common/modules/proc_removing-lifecycle-environments-from-foreman.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+= Removing Lifecycle Environments from {ProjectServer}
+
+Use this procedure to remove a lifecycle environment.
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Content* > *Lifecycle Environments*.
+. Click the name of the lifecycle environment that you want to remove, and then click *Remove Environment*.
+. Click *Remove* to remove the environment.
+
+.CLI procedure
```
Comment (maximiliankolb) [inline guides/common/modules/proc_removing-lifecycle-environments-from-foreman.adoc:11]:
```suggestion
[id="cli-Removing_Lifecycle_Environments_from_Server_{context}"]
.CLI procedure
```

File: guides/common/modules/proc_removing-lifecycle-environments-from-smart-proxy.adoc [hunk 4]
```diff
@@ -1,14 +1,14 @@
-[id="Removing_Life_Cycle_Environments_from_smart_proxy_{context}"]
-= Removing Life Cycle Environments from {SmartProxyServerTitle}
+[id="Removing_Lifecycle_Environments_from_smart_proxy_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_removing-lifecycle-environments-from-smart-proxy.adoc:1]:
```suggestion
[id="Removing_Lifecycle_Environments_from_{smart-proxy-context-titlecase}_{context}"]
```

File: guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc [hunk 5]
```diff
@@ -131,33 +131,33 @@ Such servers lie outside of the {Project} infrastructure, however, it is recomme
...
+image::life-cycle-path-simple-satellite.png[A single lifecycle environment path]
 
-* *Application specific life cycle environment paths* – each application has a separate path, which allows for individual application release cycles.
-You can associate specific compute resources with application life cycle stages to facilitate testing.
+* *Application specific lifecycle environment paths* – each application has a separate path, which allows for individual application release cycles.
+You can associate specific compute resources with application lifecycle stages to facilitate testing.
 On the other hand, this scenario increases the maintenance complexity.
 +
-image::life-cycle-path-diverged-satellite.png[Application specific life cycle environment paths]
+image::life-cycle-path-diverged-satellite.png[Application specific lifecycle environment paths]
```
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:160]:
```suggestion
image::lifecycle-path-diverged-satellite.png[Application specific lifecycle environment paths]
```

I find these images:

````
./guides/doc-Planning_for_Project/images/life-cycle-path-diverged-satellite.png
./guides/doc-Planning_for_Project/images/life-cycle-path-basic-satellite.png
./guides/doc-Planning_for_Project/images/life-cycle-path-simple-satellite.png
./guides/doc-Planning_for_Project/images/life-cycle-satellite.png
./guides/doc-Managing_Content/images/application-life-cycle-satellite.png
./guides/doc-Managing_Content/images/application-life-cycle.png
````

Could you please rename them too?

File: guides/doc-Planning_for_Project/topics/Hosts.adoc [hunk 6]
```diff
@@ -60,9 +60,9 @@ However, without inheritance there is a risk of high duplication of settings bet
 [[brid-Life_Cycle_Environment_Based_Structure]]
 *Life Cycle Environment Based Structure*
```
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Hosts.adoc:61]:
```suggestion
[[brid-Lifecycle_Environment_Based_Structure]]
*Lifecycle Environment Based Structure*
```

unrelated question: what does "brid" mean?

File: guides/doc-Planning_for_Project/topics/Org_Loc_and_Life_Cycle_Environments.adoc [hunk 7]
```diff
@@ -35,9 +35,9 @@ Each location is created and used by a single account, although each account can
 
 === Life Cycle Environments
```
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Org_Loc_and_Life_Cycle_Environments.adoc:36]:
```suggestion
=== Lifecycle Environments
```

File: guides/doc-Planning_for_Project/topics/Org_Loc_and_Life_Cycle_Environments.adoc
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Org_Loc_and_Life_Cycle_Environments.adoc:1]: Please also adjust file name, anchor, and title.

File: guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc (hunk 5 above)
Comment (bangelic) [reply guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:160]: I was able to rename the images in `doc-Planning_for_Project` but I couldn't find the images for `doc-Managing_Content`.

Comment (maximiliankolb) [reply guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc]: Maybe I was duped by symlinks???

File: guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc [hunk 8]
```diff
@@ -129,35 +129,35 @@ For detailed instructions on setting up content sources see {ContentManagementDo
 A custom repository within {ProjectServer} is in most cases populated with content from an external staging server.
 Such servers lie outside of the {Project} infrastructure, however, it is recommended to use a revision control system (such as Git) on these servers to have better control over the custom content.
 [[sect-Defining_the_Content_Life_Cycle]]
```
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:131]:
```suggestion
[[sect-Defining_the_Content_Lifecycle]]
```

**Comment by maximiliankolb** [comment]
@bangelic I believe the builds fails due to an xrefs to "Creating_an_Application_Life_Cycle_content-management".

File: guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc (hunk 5 above)
Comment (bangelic) [reply guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc]: I found them!

File: guides/common/modules/con_introduction-to-application-lifecycle.adoc [hunk 9]
```diff
@@ -0,0 +1,44 @@
+[id="introduction_to_application_lifecycle_{context}"]
+= Introduction to Application Lifecycle
+
+The _application life cycle_ is a concept central to {Project}'s content management functions.
```
Comment (maximiliankolb) [inline guides/common/modules/con_introduction-to-application-lifecycle.adoc:4]:
```suggestion
The _application lifecycle_ is a concept central to {Project}'s content management functions.
```

File: guides/doc-Planning_for_Project/topics/Introduction.adoc [hunk 10]
```diff
@@ -77,15 +77,15 @@ include::../common/modules/snip_red-hat-images.adoc[]
 endif::[]
 
 [id="figu-Content_Life_Cycle_{context}"]
```
Comment (maximiliankolb) [inline guides/doc-Planning_for_Project/topics/Introduction.adoc:79]:
```suggestion
[id="figu-Content_Lifecycle_{context}"]
```

**Review by maximiliankolb — APPROVED** [review]
LGTM!

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   d43e15b51..2b461fdde  3.9 -> 3.9
   f61988424..f8376970c  3.8 -> 3.8
   acfe650e5..db30291e2  3.7 -> 3.7

Merge conficts on 3.6:

````
	deleted by them: guides/common/assembly_managing-application-life-cycles.adoc
	deleted by us:   guides/common/modules/con_restricting-hosts-access-to-content.adoc
	both modified:   guides/common/modules/proc_adding-lifecycle-environments.adoc
	both modified:   guides/common/modules/proc_configuring-repositories.adoc
	both modified:   guides/doc-Managing_Content/docinfo.xml
	both modified:   guides/doc-Managing_Content/master.adoc
````

I am not sure if this is worth the effort. If you want to, you can cherry-pick yourself, resolve the merge conflicts, and open another PR. Your call.

**Pull request merged**